python/MooseDocs/extensions/appsyntax.py

```diff
@@ -772,6 +772,14 @@ def createMaterialize(self, parent, token, page):
         html.Tag(p, 'span', string='C++ Type:')
         html.String(p, content=cpp_type, escape=True)
 
+        doc_unit = param['doc_unit']
+        p = html.Tag(body, 'p', class_='moose-parameter-description-doc-unit')
+        html.Tag(p, 'span', string='Unit:')
+        if not doc_unit:
+            html.String(p, content='(no unit assumed for this parameter)')
```

Comment:
If it were me, I would drop "for this parameter" since that can be implied by the context of this, but I'm fine either way.